spotify_project/base/views.py

```diff
@@ -40,3 +40,8 @@ def pull_data(request):
 def all_artists(request):
     artists = Artist.objects.all().order_by(Lower('name'))
     return render(request, 'all_artists.html', {'context': artists})
+
+
+def single_artist(request, artist_id, name):
```

Comment:
In general, when you are passing along a PK in the URL, you generally want to call `get_object_or_404` on the PK. By doing this, you gracefully handle when the PK doesn't map to anything, and you also auto pull in the object you are working on. Since you didn't do this, that's likely why you are passing along the name. Once you do this, you'll be able to pass the artist into the context, then do `{{ artist.name }}`.